src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -932,10 +936,18 @@ class SelectTypeInfo:
         num_clauses: int = -1
         default_idx: int = -1
 
-    def __init__(self, modules_to_import=None):
-        self._modules_to_import = []
-        if modules_to_import:
-            self._modules_to_import = modules_to_import
+    def __init__(self, resolve_modules=False):
+        if isinstance(resolve_modules, bool):
+            self._resolve_all_modules = resolve_modules
+            self._modules_to_resolve = []
+        elif (isinstance(resolve_modules, Iterable) and
+              all(isinstance(x, str) for x in resolve_modules)):
+            self._resolve_all_modules = False
+            self._modules_to_resolve = resolve_modules
```

Comment:
For consistency, best to ensure these are all stored as lowercase.

Reply:
Done